main.go

```diff
@@ -202,16 +202,18 @@ func run(conf configuration, listen string, debug bool) {
 	// Initialize a monitor.
 	monitor := newMonitor(ag, conf.MonitorInterval)
 
+	stats := ag.stats
+
 	router := mux.NewRouter().StrictSlash(true)
-	router.HandleFunc("/__heartbeat__", ag.handleHeartbeat).Methods("GET")
-	router.HandleFunc("/__lbheartbeat__", handleLBHeartbeat).Methods("GET")
-	router.HandleFunc("/__version__", handleVersion).Methods("GET")
-	router.HandleFunc("/__monitor__", monitor.handleMonitor).Methods("GET")
-	router.HandleFunc("/sign/files", ag.handleSignature).Methods("POST")
-	router.HandleFunc("/sign/file", ag.handleSignature).Methods("POST")
-	router.HandleFunc("/sign/data", ag.handleSignature).Methods("POST")
-	router.HandleFunc("/sign/hash", ag.handleSignature).Methods("POST")
-	router.HandleFunc("/auths/{auth_id:[a-zA-Z0-9-_]{1,255}}/keyids", ag.handleGetAuthKeyIDs).Methods("GET")
+	router.HandleFunc("/__heartbeat__", statsMiddleware(ag.handleHeartbeat, "http.heartbeat", stats)).Methods("GET")
+	router.HandleFunc("/__lbheartbeat__", statsMiddleware(handleLBHeartbeat, "http.lbhearbeat", stats)).Methods("GET")
+	router.HandleFunc("/__version__", statsMiddleware(handleVersion, "http.__version", stats)).Methods("GET")
+	router.HandleFunc("/__monitor__", statsMiddleware(monitor.handleMonitor, "http.monitor", stats)).Methods("GET")
```

Comment:
Do we really need to send stats for _all_ of these endpoints? `/__heartbeat__` seems like all we would need to measure uptime. (I guess `/__lbheartbeat__` as well if we care about measuring full uptime vs. app up but db down uptime.)

Reply:
Oh, "uptime" in my prior experience has changed to mean more like "availability", or "success rate", and isn't really about whether the service is "running".

These are heartbeats so that we know the service has come up correctly during deploy and so that the service has traffic routed to it by the loadbalancers. But they don't tell us much about how successfully we're handling user requests, so we need the actual counts from the API calls.

Reply:
But I think I'm just sloppy with it. I def meant "availability" or "success rate"

Reply:
If I understand correctly, you're saying that what we really care about is the counts from the actual API that distribution services use, which makes sense to me. I guess I'm still not clear on why it's worth instrumenting these endpoints though? And especially why we're instrumenting `__version__`. It's not a major objection, just curious.

Reply:
The heartbeat's to help discover problems with them (so we can figure out why an lb is dropping the machine: did it reach the process and it was erroring or was it pointing at a process that was already down?) and "do all the same things" for the `__version__` one. Just in case we, I dunno, kept crashing the process whenever __version__ got hit or something.